vm/systemSmartContracts/staking.go

```diff
@@ -595,6 +611,11 @@ func (r *stakingSC) unStake(args *vmcommon.ContractCallInput) vmcommon.ReturnCod
 		r.eei.AddReturnMessage("unStake is not possible for jailed nodes")
 		return vmcommon.UserError
 	}
+	err = r.moveFirstFromWaitingToStakedIfNeeded(args.Arguments[0])
+	if err != nil {
```

Comment:
This is not really ok, the move from waiting failed because of some other registrationData not because of userError.

Reply:
this should not happen, only in case of programming error. if that happens, better leave the contracts state as it is.

Reply:
ok